### PR TITLE
Ignore Sourcepoint cookie paywall

### PR DESCRIPTION
**Pull request by sammacbeth**
https://app.asana.com/0/0/1203288491490638/f

Mark popup as not open if it has only 'subscribe or accept all' options.

### CODE REVIEW

File: lib/cmps/sourcepoint-frame.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ export default class SourcePoint extends AutoConsentCMPBase {
   }
 
   async detectPopup() {
-    return true;
+    // check for the paywall button, and bail if it exists to prevent broken opt out
+    await waitForElement(".sp_choice_type_11,.sp_choice_type_12,.sp_choice_type_13,.sp_choice_type_ACCEPT_ALL", 2000);
+    return !elementExists('.sp_choice_type_9');
```
Comment (muodov) [inline lib/cmps/sourcepoint-frame.ts:41]: Will the opt-in mode still work?

File: tests/sourcepoint.spec.ts [hunk 2]
```diff
@@ -15,3 +13,16 @@ generateCMPTests('Sourcepoint-frame', [
...
+generateCMPTests('Sourcepoint-frame', [
+    "https://www.sueddeutsche.de/",
+    "https://www.derstandard.at/",
+    "https://www.heise.de/",
+    'https://www.n-tv.de/',
+], {
+    skipRegions: ["US", "GB", "FR"],
+    expectPopupOpen: false,
+    testOptIn: false,
+    testOptOut: false,
```
Comment (muodov) [inline tests/sourcepoint.spec.ts:26]: If both testOptout and testOptin are off, will the test ever be run?

File: lib/cmps/sourcepoint-frame.ts (hunk 1 above)
Comment (sammacbeth) [reply lib/cmps/sourcepoint-frame.ts:41]: No. For sites using the subscription button, we just consider the popup closed, and both opt-in and opt-out won't happen. Not sure that there is another way to handle this without making changes to the API.

For the meantime, I think it's worth making this change, so we don't fail with opt-outs on these sites.

File: tests/sourcepoint.spec.ts (hunk 2 above)
Comment (sammacbeth) [reply tests/sourcepoint.spec.ts:26]: We will still test for CMP detection and whether the popup is open.

**Pull request merged**